packages/sdk/src/cosmwasmclient.ts

```diff
@@ -0,0 +1,228 @@
+      sequence: account.sequence,
+    };
+  }
+
+  public async postTx(tx: Uint8Array): Promise<PostTxResult> {
+    const result = await this.restClient.postTx(tx);
+    if (result.code) {
+      throw new Error(`Error when posting tx. Code: ${result.code}; Raw log: ${result.raw_log}`);
+    }
+    return {
+      logs: parseLogs(result.logs) || [],
+      rawLog: result.raw_log || "",
+      transactionHash: result.txhash,
+    };
+  }
+
+  /** Uploads code and returns a code ID */
+  public async upload(wasmCode: Uint8Array, memo = ""): Promise<number> {
+    const storeCodeMsg: MsgStoreCode = {
+      type: "wasm/store-code",
+      value: {
+        sender: this.senderAddress,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        wasm_byte_code: Encoding.toBase64(wasmCode),
+        source: "",
+        builder: "",
+      },
+    };
+    const fee = defaultUploadFee;
+    const { accountNumber, sequence } = await this.getNonce();
+    const chainId = await this.chainId();
+    const signBytes = makeSignBytes([storeCodeMsg], fee, chainId, memo, accountNumber, sequence);
+    const signature = await this.signCallback(signBytes);
+    const signedTx = {
+      msg: [storeCodeMsg],
+      fee: fee,
+      memo: memo,
+      signatures: [signature],
+    };
+
+    const result = await this.postTx(marshalTx(signedTx));
+    const codeIdAttr = findAttribute(result.logs, "message", "code_id");
+    const codeId = Number.parseInt(codeIdAttr.value, 10);
+    return codeId;
+  }
+
+  public async instantiate(
+    codeId: number,
+    initMsg: object,
+    memo = "",
+    transferAmount?: readonly Coin[],
+  ): Promise<string> {
+    const instantiateMsg: MsgInstantiateContract = {
+      type: "wasm/instantiate",
+      value: {
+        sender: this.senderAddress,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        code_id: codeId.toString(),
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        init_msg: initMsg,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        init_funds: transferAmount || [],
+      },
+    };
+    const fee = defaultInitFee;
+    const { accountNumber, sequence } = await this.getNonce();
+    const chainId = await this.chainId();
+    const signBytes = makeSignBytes([instantiateMsg], fee, chainId, memo, accountNumber, sequence);
+
+    const signature = await this.signCallback(signBytes);
+    const signedTx = {
+      msg: [instantiateMsg],
+      fee: fee,
+      memo: memo,
+      signatures: [signature],
+    };
+
+    const result = await this.postTx(marshalTx(signedTx));
+    const contractAddressAttr = findAttribute(result.logs, "message", "contract_address");
+    return contractAddressAttr.value;
+  }
+
+  public async execute(
+    contractAddress: string,
+    handleMsg: object,
+    memo = "",
+    transferAmount?: readonly Coin[],
+  ): Promise<{ readonly logs: readonly Log[] }> {
```

Comment:
Yeah more flexible than just `readonly Log[]`. One day I will figure out how to get the data value here

Reply:
Good point. I create the type `ExecuteResult` for easier extensibility in 90e76b7a41996c459a9665dda049d22918bc4565